doc/source/user/build_system/core_files.rst

```diff
@@ -0,0 +1,257 @@
+To get started, here's the full ``blinky.core`` file.
+The following sections will refer back to this example to discuss it in detail.
+
+.. literalinclude:: ../../../../tests/userguide/blinky/blinky.core
+   :language: yaml
+   :caption: ``blinky.core``, an exemplary core file
+   :name: ug_build_system_cb_blinky_full
+   :linenos:
+
+Naming the core file
+--------------------
+
+The core file can have any name, but it must end in ``.core``.
+It is recommended to choose a file name matching the core name, as discussed below.
+
+The first line: ``CAPI=2``
+--------------------------
+
+A core file always starts with the line ``CAPI=2``.
+No other content (including comments) is allowed before this line, as FuseSoC uses this line to differentiate between different versions of the CAPI schema.
+Only CAPI version 2 is specified at the moment.
+
+.. _ug_build_system_core_name:
+
+The core name, version, and description
+---------------------------------------
+
+Each core has a name, given in the ``name`` key.
+Core names can be freely chosen, but need to follow a common structure called *VLNV*.
+:term:`VLNV` stands the four parts of a core name, which are separated by colon (``:``): Vendor, Library, Name, and Version.
+
+.. todo::
+
+   Specify version number requirements, and what happens if no version number is given.
+
+Cores can also have a description, given in the ``description`` key.
+A description is optional, but recommended.
+
+.. literalinclude:: ../../../../tests/userguide/blinky/blinky.core
+   :language: yaml
+   :start-at: name:
+   :end-at: description:
+
+In this example, the vendor is ``fusesoc``, the library is ``examples``, and the name of the core is ``blinky``.
+The version is set to ``1.0``.
+
+Specifying source files
+-----------------------
+
+A core typically consists of one or multiple source files.
+Source files are grouped into file sets under the ``filesets`` key.
+
+FuseSoC does neither mandate a specific grouping, nor naming of file sets.
+It is common to use one file set for RTL (design) files, and one for testbench files.
+
+The following example shows a single file set, ``rtl``, with a set of common keys.
+
+.. literalinclude:: ../../../../tests/userguide/blinky/blinky.core
+   :language: yaml
+   :start-at: filesets:
+   :end-at: file_type
+
+For each named file set, several keys are supported:
+
+* ``files``: An ordered list of source files.
+  The list of source files is ordered: the files will be passed to the tool in exactly the given order.
+  This is important, for example, in SystemVerilog, where packages need to be compiled before they can be used by subsequent source files.
+* ``file_type``: The default file type for all files in the ``files`` list.
+* ``depend``: Dependencies on other cores. More on dependencies later.
+
+Source files
+~~~~~~~~~~~~
+
+Source files are resolved relative to the location of the core file and must be stored in the same directory as the core file, or in a subdirectory of it.
```

Comment:
Should we mention already here that remote cores work somewhat differently?

Reply:
I have ignored the package manager-like topics so far and haven't really thought about how to best phrase things to avoid confusion. I'd leave things as-is for now, and revisit when I do the package manager parts.